packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.java

```diff
@@ -52,8 +53,21 @@
     void onSuccess(AssetEntity assetEntity, boolean isNew);
   }
 
-  public static void downloadFileToPath(Request request, final File destination, final FileDownloadCallback callback) {
-    downloadData(request, new Callback() {
+  private static OkHttpClient getClient(Context context) {
+    if (sClient == null) {
+      sClient = new OkHttpClient.Builder().cache(getCache(context)).build();
```

Comment:
I'm a bit worried about saving a context-specific client in a static variable. In this case using a `WeakHashMap<Context, OkHttpClient>` doesn't seem right but perhaps it'd help to add a runtime assertion that context.getCacheDir() matches sClient.cache().directory().getAbsolutePath().

Reply:
OK, I think that makes sense, will follow up with a separate PR in a bit.